### PR TITLE
Feat/functional api call

### PR DESCRIPTION
**Pull request by CapCinematic**
Did this break anything?
- [x] No
- [ ]  Yes
Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x]  Bug fix (non-breaking change which fixes an issue)
- [x]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.
Checklist:
- [ ]  If this code needs to be tested, all tests are passing
- [x]  The code runs locally: You may need to add a Chrome extension to allow CORS
- [ ]  There are comments where clarification/ organization is needed
- [ ]  The code is DRY. If it's not, I tried my best
- [ ]  I reviewed my code before pushing
Summary:
Race and I fixed a CORS issue when trying to fetch data from our API by adding a CORS extension from the Google extension store. In this branch, we also refactored our Type definition for our bracket data defining proper objects and strings. The bracket now uses our API and you can make it all the way through with no errors.

### CODE REVIEW

**Comment by SethBurns** [comment]
We have build tests failing in Circle CI. Did we merge main into this branch before pushing?

**Comment by epintozzi** [comment]
Aside from CI failing this looks good. I think BE might also want to make some changes regarding the CORS issue. It's not really an ideal solution to require users to install a browser extension (if I'm understanding your solution correctly). I suspect you know that though and this is just an alternate workaround for the time being.

Other thoughts: There's 2 (mostly) unrelated things happening in this PR. Ideally there'd be 1 PR to address the CORS issue and a separate PR to make the updates needed for using the BE API. That might mean a really tiny PR for the CORS thing but that's okay. I realize you need the CORS thing fixed for the rest to fully work but they are still technically different problems being solved. Not a huge deal but it's not a habit we want to be in either because it's the kind of thing that can easily snowball. 

I'm also curious if there is a reasonable way to make this more dynamic so you don't need to manually address each bracket size. I don't think that is something that needs to be changed for this PR, but it's worth thinking about how you might handle that. I could see that being a reasonable question that comes up if this were an interview code challenge type situation.

**Comment by epintozzi** [comment]
Marking as approved assuming CI is resolved before merging

**Review by epintozzi — APPROVED** [review]

**Review by SethBurns — APPROVED** [review]

**Pull request merged**
